sdk/search/azure-search/src/main/java/com/azure/search/SearchServiceAsyncClient.java

```diff
@@ -0,0 +1,1759 @@
+     * @return true if the index exists; false otherwise.
+     */
+    public Mono<Response<Boolean>> indexExistsWithResponse(String indexName, RequestOptions requestOptions) {
+        try {
+            return withContext(context -> this.indexExistsWithResponse(indexName, requestOptions, context));
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    Mono<Response<Boolean>> indexExistsWithResponse(String indexName,
+                                                    RequestOptions requestOptions,
+                                                    Context context) {
+        return resourceExistsWithResponse(() -> this.getIndexWithResponse(indexName, requestOptions, context));
+    }
+
+    /**
+     * Returns statistics for the given index, including a document count and storage usage.
+     *
+     * @param indexName the name of the index for which to retrieve statistics
+     * @return the index statistics result.
+     */
+    public Mono<GetIndexStatisticsResult> getIndexStatistics(String indexName) {
+        try {
+            return this.getIndexStatisticsWithResponse(indexName, null)
+                .map(Response::getValue);
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    /**
+     * Returns statistics for the given index, including a document count and storage usage.
+     *
+     * @param indexName the name of the index for which to retrieve statistics
+     * @param requestOptions additional parameters for the operation.
+     *                       Contains the tracking ID sent with the request to help with debugging
+     * @return a response containing the index statistics result.
+     */
+    public Mono<Response<GetIndexStatisticsResult>> getIndexStatisticsWithResponse(String indexName,
+                                                                                   RequestOptions requestOptions) {
+        try {
+            return withContext(context -> this.getIndexStatisticsWithResponse(indexName, requestOptions, context));
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    Mono<Response<GetIndexStatisticsResult>> getIndexStatisticsWithResponse(String indexName,
+                                                                            RequestOptions requestOptions,
+                                                                            Context context) {
+        return restClient
+            .indexes()
+            .getStatisticsWithRestResponseAsync(indexName, requestOptions, context)
+            .map(Function.identity());
+    }
+
+    /**
+     * Lists all indexes available for an Azure Cognitive Search service.
+     *
+     * @return a reactive response emitting the list of indexes.
+     */
+    public PagedFlux<Index> listIndexes() {
+        try {
```

Comment:
@navalev If I understand this correctly, this method is delegating to the other overload of `listIndexes` that already has the `try`/`catch` block -- Is it still needed here as well?

Reply:
All the public APIs are using try/catch block. Although it is calling a method that handles this as well. We aim to follow the same structure for all the public APIs. If you see this as a mandatory change, I suggest doing this in a seperate PR and re-running all tests